airbyte-workers/src/main/java/io/airbyte/workers/storage/GcsCloudDocumentStoreClient.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.storage;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * Document store on top of the GCS Client (Storage).
+ */
+public class GcsCloudDocumentStoreClient implements CloudDocumentStoreClient {
+
+  private final String bucketName;
+  private final Path root;
+  private final Storage gcsClient;
+
+  public GcsCloudDocumentStoreClient(final Storage gcsClient, final String bucketName, final Path root) {
+    this.gcsClient = gcsClient;
+    this.bucketName = bucketName;
```

Comment:
Should we create the bucket if it doesn't exist? That will help with the first run of this.

Reply:
oh. good point. yes i can add that.

Reply:
actually @jrhizor & @davinchia how did you think about this for `CloudLogs`? It looks like it assumes that the buckets do exist (and I assume fails if they don't).

Reply:
Yeap. I don't think we should have this create buckets. I think it's simpler if we have the expectation that configuration passed into the cluster is expected to be correct. 

This means:
1) the GCP creds have reduced permissions, which is good
2) any one operating Airbyte is forced to properly managed their buckets, which is also good.

for these reasons, I don't think this adds that much value.

Reply:
Agree with @davinchia here 100%